File: Cs/Solution/IsBalanced.cs

namespace Solution
{
	public partial class Solution
	{
		public bool IsBalanced(string num)
		{
			int leftSum = 0;
			int rightSum = 0;
			int n = num.Length;
			for (int i = 0; i < n; i += 2)
			{
				leftSum += num[i] - '0';
			}
			for (int i = 1; i < n; i += 2)
			{
				rightSum += num[i] - '0';
			}
			return leftSum == rightSum;
		}
	}
}

File: Rust/src/solution/is_balanced.rs
use super::Solution;

impl Solution {
    pub fn is_balanced(num: String) -> bool {
	let bytes = num.as_bytes();
	let n = bytes.len();
	let mut left_sum = 0;
	let mut right_sum = 0;
	for i in (0..n).step_by(2) {
	    left_sum += (bytes[i] - b'0') as i32;
	}
	for i in (1..n).step_by(2) {
	    right_sum += (bytes[i] - b'0') as i32;
	}
	left_sum == right_sum
    }
}
